README.md

```diff
@@ -29,6 +29,30 @@ well.
 Run `tsickle --help` for the full syntax, but basically you provide any tsickle
 specific options and use it as a TypeScript compiler.
 
+### Differences from TypeScript
+
+Closure and TypeScript are not identical.  Tsickle hides most of the
+differences, but users must still be aware of some differences.
+
+#### `declare`
+
+Any declaration in a `.d.ts` file, as well as any declaration tagged with
+`declare ...`, is intepreted by Tsickle as a name that should be preserved
+through Closure compilation (i.e. not renamed into something shorter).  Use it
+any time the specific string names of your fields are important.
+
+Example:
+
+    declare interface JSONResult {
+        username: string;
+    }
+    let r = JSON.parse(input) as JSONResult;
```

Comment:
There is also a co-example when the object is constructed in the TypeScript code but passed out of the compilation scope.

```
let r = {username: ...};
window.fetch(..., r);
```